Python/optimizer_bytecodes.c

```diff
@@ -1216,6 +1216,18 @@ dummy_func(void) {
         sym_set_const(callable, list_append);
     }
 
+    op(_BINARY_SLICE, (container, start, stop -- res)) {
+        // Slicing a string/list/tuple always returns the same type.
+        PyTypeObject *type = sym_get_type(container);
+        if (type == &PyUnicode_Type ||
+            type == &PyList_Type ||
```

Comment:
```suggestion
            type == &PyBytes_Type ||
            type == &PyList_Type ||
```
I do not know whether slicing a bytes object occurs often enough to add `PyBytes_Type` here (and whether the jit slows down if we add more cases here), but if I understand correctly slicing `bytes`  (or `bytearray`) results in a `bytes` (or `bytearray`) so we could add them.

Reply:
@fluhus, this PR is probably fine for now, but feel free to add other sequence types like this in a follow-up PR (with tests)!